pylint_odoo/checkers/modules_odoo.py

```diff
@@ -368,7 +368,8 @@ def check_folder_test_imported(self, node):
         if (hasattr(node.parent, 'file')
             and os.path.basename(node.parent.file) == '__init__.py'
             and ("tests" in [item for item, _ in node.names]
-                 or "tests" in node.modname)):
+                 or (hasattr(node, 'modname')
+                    and "tests" in node.modname))):
```

Comment:
Pro tip:  
Instead of:  
`hasattr(node, 'modname and 'test' in node.modname`  

It may be simplified as:  
`'test' in getattr(node, 'modename', '')`

Similar as dictionary, you may use a getter with a default method.

Reply:
A import node could be a `import package` or `from . import package`

It has 2 different classes:
 - astroid.ImportFrom
 - astroid.Import

You are just validating the case:
  - `from . import tests`

So, you can use the following validation:
 - https://github.com/OCA/pylint-odoo/blob/b4dae33082192c7a0b85a19cf9e7c68e35516581/pylint_odoo/checkers/no_modules.py#L448-L449

Or if you want to support the following case:
  - `import tests`

You can se the following validation:
 - https://github.com/OCA/pylint-odoo/blob/b4dae33082192c7a0b85a19cf9e7c68e35516581/pylint_odoo/checkers/no_modules.py#L450-L451

This code will be compatible with multi-imports lines like
  - `from . import tests, models`
  -  `import tests, models`
 
Regargs!